packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -2285,7 +2287,19 @@ class RenderEditable extends RenderBox with RelayoutWhenSystemFontsChangeMixin,
     switch (defaultTargetPlatform) {
       case TargetPlatform.iOS:
       case TargetPlatform.macOS:
-        _caretPrototype = Rect.fromLTWH(0.0, 0.0, cursorWidth, cursorHeight + 2);
+        final double styleHeight = _textPainter.text?.style?.height ?? 1.0;
+        // Extend the cursor height only when the text style height is less than 1.3.
+        // Extending the cursor height made sense on M2 whose default text style had
+        // a 1.0 line height. On M3, whose default line height is 1.5, the extension
+        // looks bad and might overflow on the label.
+        // When line height is equal or superior to 1.3, cursor height is reduced by
+        // a small amount to get a better rendering.
+        // TODO(bleroux): replace this fix with a long term solution, see
+        // https://github.com/flutter/flutter/issues/7034 and
+        // https://github.com/flutter/flutter/issues/120836.
```

Comment:
Yeah it seems like there should be a better solution here, but maybe this fix is worth it for now. We should be sure to check any golden or Google tests that might be broken by this.